azurerm/internal/services/policy/policy_set_definition_resource.go

```diff
@@ -420,16 +437,25 @@ func policySetDefinitionRefreshFunc(ctx context.Context, client *policy.SetDefin
 	}
 }
 
-func expandAzureRMPolicySetDefinitionPolicyDefinitions(input []interface{}) *[]policy.DefinitionReference {
+func expandAzureRMPolicySetDefinitionPolicyDefinitions(input []interface{}) (*[]policy.DefinitionReference, error) {
 	result := make([]policy.DefinitionReference, 0)
 
 	for _, item := range input {
 		v := item.(map[string]interface{})
 
 		parameters := make(map[string]*policy.ParameterValuesValue)
-		for k, value := range v["parameters"].(map[string]interface{}) {
-			parameters[k] = &policy.ParameterValuesValue{
-				Value: value.(string),
+		if p, ok := v["parameter_values"]; ok {
+			_ = json.Unmarshal([]byte(p.(string)), &parameters) // this is ensured by schema validation
```

Comment:
we should confirm/raise this error here anyway - since bugs can happen

Reply:
Sure